tests/008-test-change-lv-data-size.sh

```diff
@@ -0,0 +1,61 @@
+source $SRCDIR/libtest.sh
+
+# Test that the DATA_SIZE option for d-s-s actually 
+# changes the size of the docker data LV carved from the VG.
+test_change_lv_data_size() {
+  local devs=$TEST_DEVS
+  local test_status=1
+  local testname=`basename "$0"`
+  local vg_name="dss-test-foo"
+  local data_size=70
+  local pe_total pe_free pe_used_percentage
+
+  if vg_exists "$vg_name"; then
+    echo "ERROR: $testname: Volume group $vg_name already exists" >> $LOGS
+    return $test_status
+  fi
+  
+  cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS="$devs"
+VG=$vg_name
+DATA_SIZE=$data_size%FREE
+EOF
+ 
+  # Run docker-storage-setup
+  $DSSBIN >> $LOGS 2>&1
+
+  # DSS execution failed.
+  if [ $? -ne 0 ]; then
+    cleanup $vg_name "$devs"
+    return $test_status
+  fi
+
+  # Error out if VG wasn't created
+  if ! vg_exists "$vg_name"; then
+    echo "ERROR: $testname: Volume group $vg_name failed to be created by DSS" >> $LOGS
+    cleanup $vg_name "$devs"
+    return $test_status
+  fi
+ 
+  # Physical extent information about the VG
+  pe_total=$(vgdisplay --noheading -C -o vg_extent_count $vg_name)
+  pe_free=$(vgdisplay --noheading -C -o vg_free_count $vg_name)
+  pe_used_percentage=$(( (pe_total - pe_free) * 100 / pe_total))
+  
+  # Give some room for error since some rounding may have occured
+  local lower_bound=$(( data_size - 3 ))
+  local upper_bound=$(( data_size + 2 ))
+
```

Comment:
Instrad of doing 5% margin of error, can we calculate available PE in volume group before thin pool creation, and then calculate 70% of that and then make sure used PEs are same as calculated. If 70% is not a integer number then lvm must be either rounding it up or down. So say 70% comes out to be 213.3. Then either 213 or 214 can both the right answer for number of PEs used after thin pool creation.